src/reducers/FiltersReducer.js

```diff
@@ -0,0 +1,58 @@
+const ensureFilterSelected = (existingFilters) => {
+  if (existingFilters.length === 0) {
+    return existingFilters
+  }
+  if (existingFilters.find(f => f.selected)) {
+    return existingFilters
+  }
+
+  const first = Object.assign({}, existingFilters[0], { selected: true })
+  const rest = existingFilters.slice(1)
+  return [first, ...rest]
+}
+
+const filtersUpdate = (existingFilters, { filter }) => {
+  let updatedFilters = null
+
+  if (!existingFilters.find(f => f.name === filter.name)) {
```

Comment:
Can you swap the order of the if/else so we don't start with a negative condition?